frontend/src/features/public-form/components/FormFields/FormFields.tsx

```diff
@@ -74,8 +80,11 @@ export const FormFields = ({
   }, [formFields, searchParams])
 
   const augmentedFormFields = useMemo(
-    () => formFields.map(augmentWithMyInfo),
-    [formFields],
+    () =>
+      formFields
+        .map(augmentWithMyInfo)
+        .map(augmentWithWorkflowDisabling.bind(this, workflowStep)),
```

Comment:
updated in [17083cf](https://github.com/opengovsg/FormSG/pull/7091/commits/17083cf46976ea39ff4e3b51623c617993e301e9)

Actually here i was a bit annoyed at how i have to give a variable name haha. Basically the idea is I want a partially-applied function, with only the first argument provided. There are three ways of doing it

1. Write the function with two parameters, and create an anonymous function inline (this is the way i did it in the updated commit, above)
```
// Define
const augmentWithWorkflowDisabling = (workflowStep, formFields) => ...

// Call
formFields.map((fields) => augmentWithWorkflowDisabling(workflowStep, fields))
```
The advantage is this is all very nice and maintainable, we're used to functions with multiple parameters, but is annoying because i shouldn't really need to define a new variable and think about it's name and what it means just to pass it in to a function and never do anything else interesting with it.

2. Write the function with two parameters, and bind it to create a partially applied function (this is the way I did it previously)
```
// Define
const augmentWithWorkflowDisabling = (workflowStep, formFields) => ...

// Call
formFields.map(augmentWithWorkflowDisabling.bind(this, workflowStep))
```
In this case, this is nice because I've removed the need to define a new variable, but now i include this random reference to ".bind" which no one really understands, and references a context "this" which doesn't exist (it's the global module context, which is a bit weird). Also not ideal

3. Curry the function, and partially apply it as usual (my preferred way, but not typical coding pattern in FormSG)
```
// Define
const augmentWithWorkflowDisabling = (workflowStep) => (formFields) => ...

// Call
formFields.map(augmentWithWorkflowDisabling(workflowStep))
```
This is **very nice** when used in this scenario, solves all the problems and is easy to read. But, using curried functions aren't very typical in the codebase, and will look weird if used in other places where partial application is not necessary (e.g. if someone else down the line needs to use `augmentWithWorkflowDisabling` but has both `workflowStep` and `formFields` available in the surrounding context, the call signature becomes `augmentWithWorkflowDisabling(workflowStep)(formFields)`, which can look confusing in a sea of typically-uncurried functions).

In the end I think just going with 1 makes the most sense and is most consistent with the rest of the codebase, even though i like 3 the most hahaha

Reply:
https://ramdajs.com/docs/#curry helps for 3. But ramda isn't a library that FormSG uses.

I notice that lodash also supports [placeholders](https://lodash.com/docs/4.17.15#curry) in curried functions. But this can be an exercise for the next time.

I've seen 1/3, but never 2 😆.

Reply:
If i have to write `.curry` then that feels a bit like, it defeats the purpose of currying anyway though. HAHAHA

Reply:
re: ramda, What?? what is this javascript magic that allows you to write `f(1)(2,3)` or `f(1,2)(3)` tf